### DISCUSSION TITLE
Edit Toot Time Limit

### DISCUSSION THREAD

**Discussion by ghost** [mastodon/mastodon#18238] [General]
Now that editing Toots has rolled out, it might be worth adding a time limit to the ability to edit.

For example, default it so a user can only edit their post for thirty minutes after posting. After the thirty minutes is up, the post can no longer be edited. The server admin should probably be able to edit the time limit.

I don't know if it's a good idea or not, it just seems like something that could be added to curb potential abuse

**Comment by linux-lukas** [comment]
Couldn't you implement a traceability, like it is given with the version management tool "git" for this function?

**Reply by ClearlyClaire** [reply]
An history of known edits is recorded by Mastodon and made available through the REST API and the web interface

**Reply by linux-lukas** [reply]
So that means it would potentially be easy to show some kind of change history of edited toots because Mastodon recognises this information, right?😅

**Reply by ClearlyClaire** [reply]
As I just wrote, it *is* shown in the web interface. See https://github.com/mastodon/mastodon/pull/17390

**Reply by linux-lukas** [reply]
Ah, oh well sorry. 

From Mastodon version 3.5 and above, right? That seems to be the reason why I don't see it yet on the instance where I live (running on 3.4.6). :D

**Reply by linux-lukas** [reply]
Could the changes be colour-coded with red for what has been deleted and green for what has been added?

**Reply by linux-lukas** [reply]
Similar as known from GitLab
